app/utils.ts

```diff
@@ -87,3 +89,13 @@ export const generateRandomNumberString = () => {
 
   return randomString;
 };
+
+export const validateNavigationRedirect = (navigation: Navigation) => {
+  return (
+    navigation.state === "loading" &&
+    navigation.formMethod !== null &&
+    navigation.formMethod !== "GET" // &&
```

Comment:
Locally I was getting this type error: `This comparison appears to be unintentional because the types 'FormMethod | undefined' and '"GET"' have no overlap.ts(2367)`

And indeed, when inspecting FormMethod, I see:

```typescript
export declare type MutationFormMethod = "post" | "put" | "patch" | "delete";
export declare type FormMethod = "get" | MutationFormMethod;
```

So changing to this fixes the error:

```suggestion
    navigation.formMethod !== "get" // &&
```

Reply:
Adding `v2_normalizeFormMethod: true,` to `remix.config.js` also fixes this error. I like that approach more.